src/opers.c

```diff
@@ -357,64 +362,74 @@ Obj FuncSIZE_FLAGS (
 **
-*F  FuncIS_EQUAL_FLAGS( <self>, <flags1>, <flags2> )  equality of flags lists
+*F  EqFlags( <flags1>, <flags2> ) . . . . . . . . . . equality of flags lists
 */
-Obj FuncIS_EQUAL_FLAGS (
-    Obj                 self,
-    Obj                 flags1,
-    Obj                 flags2 )
+Int EqFlags(Obj flags1, Obj flags2)
 {
     Int                 len1;
     Int                 len2;
     UInt  *             ptr1;
     UInt  *             ptr2;
     Int                 i;
 
-    /* do some trivial checks                                              */
-    while ( TNUM_OBJ(flags1) != T_FLAGS ) {
-        flags1 = ErrorReturnObj( "<flags1> must be a flags list (not a %s)",
-            (Int)TNAM_OBJ(flags1), 0L,
-            "you can replace <flags1> via 'return <flags1>;'" );
-    }
-    while ( TNUM_OBJ(flags2) != T_FLAGS ) {
-        flags2 = ErrorReturnObj( "<flags2> must be a flags list (not a %s)",
-            (Int)TNAM_OBJ(flags2), 0L,
-            "you can replace <flags2> via 'return <flags2>;'" );
-    }
-    if ( flags1 == flags2 ) {
-        return True;
-    }
-
-    /* do the real work                                                    */
+    // do the real work
     len1 = NRB_FLAGS(flags1);
     len2 = NRB_FLAGS(flags2);
     ptr1 = BLOCKS_FLAGS(flags1);
     ptr2 = BLOCKS_FLAGS(flags2);
     if ( len1 <= len2 ) {
         for ( i = 1; i <= len1; i++ ) {
             if ( *ptr1 != *ptr2 )
-                return False;
+                return 0;
             ptr1++;  ptr2++;
         }
         for ( ; i <= len2; i++ ) {
             if ( 0 != *ptr2 )
-                return False;
+                return 0;
             ptr2++;
         }
     }
     else {
         for ( i = 1; i <= len2; i++ ) {
             if ( *ptr1 != *ptr2 )
-                return False;
+                return 0;
             ptr1++;  ptr2++;
         }
         for ( ; i <= len1; i++ ) {
             if ( *ptr1 != 0 )
-                return False;
+                return 0;
             ptr1++;
         }
     }
-    return True;
+    return 1;
+}
+
+
+/****************************************************************************
+**
+*F  FuncIS_EQUAL_FLAGS( <self>, <flags1>, <flags2> )  equality of flags lists
+*/
+Obj FuncIS_EQUAL_FLAGS (
+    Obj                 self,
+    Obj                 flags1,
+    Obj                 flags2 )
+{
+    /* do some trivial checks                                              */
+    while ( TNUM_OBJ(flags1) != T_FLAGS ) {
+        flags1 = ErrorReturnObj( "<flags1> must be a flags list (not a %s)",
+            (Int)TNAM_OBJ(flags1), 0L,
+            "you can replace <flags1> via 'return <flags1>;'" );
+    }
+    while ( TNUM_OBJ(flags2) != T_FLAGS ) {
+        flags2 = ErrorReturnObj( "<flags2> must be a flags list (not a %s)",
+            (Int)TNAM_OBJ(flags2), 0L,
+            "you can replace <flags2> via 'return <flags2>;'" );
+    }
+    if ( flags1 == flags2 ) {
```

Comment:
Why don't you do this check in the `EqFlags` method?

Reply:
Because this is an internal API, where the caller is expected to already have checked the input. And when called via EQ / EqFuncs, the tnums are automatically verified, too.

Reply:
Yeah, I think I put the comment in a slightly confusing place, and of course did not specify what I really meant, sorry.

What I actually meant was:
Why is the `if ( flags1 == flags2 )` check not in the `EqFlags` method?

Reply:
Ahhh, sorry, I overlooked that line. Good question, I guess I forgot to move it, oops.

Reply:
No worries :). I think I should have specified this more.